Update/v15 - New Timelines chapter (#295)
AvolitesLtd/TitanManual: docs/timelines.md

---
id: timelines
title: Timelines introduction
sidebar_label: Timelines introduction
---

Timelines allow you to create a precisely timed light show which you can either
run from the console's internal timer or synchronise to an external timecode
source. This can be useful for an opening sequence for a corporate
event, a complex song in a show where the song is pre-recorded or run to 

> a click track or timecode, or a multimedia event run to a playback track.

Reviewer: We also added a way of using it for one shots, although that isn't the main usage. Not sure if that's worth a mention here or not really.

You can set the fader to automatically play the timeline and pause/stop on zero. Using internal timecode you can use this to create some complicated effects that you can use while busking.

> You can also create timed shows by creating timecoded cues in 
[cue-lists](./cue-lists/cue-list-timing.md#running-a-cue-list-to-timecode), but Timeline 
recording allows much easier creation and editing of a timed show.

Timelines can be built either from cues or chases you have already recorded, or by creating
new cues as you go along. You can record them in real time by operating the console
or by adding cues manually at specific times. You can think of a timeline as an automatic replay
system to push the faders and buttons on the console. The actions recorded in a timeline
are called **triggers**.

To show the Timeline View window for a playback containing a timeline, touch the playback 
information area relating to the fader or press \<Open/View\> then the **select button** 
of the playback.

![Playback View of Timeline](/docs/images/Timeline-Window.png)

Triggers
------------------

> To program a timeline you record actions, or **Triggers**, at specific times

Reviewer: You record actions at specific times. A trigger has a time and an action if that makes sense?

To set up and enable each of the timecode sources as Internal, MIDI, Clock or SMPTE use the \[Timecode\] softkey at the top level main menu.

Local timecode controls
------------------

The \{Link\} button in the bottom left of the Timeline View window links or unlinks the timeline with the timecode source. Especially with external timecode, it's useful when recording and editing to be able to control the timecode yourself rather than having to continually ask the video guys to rewind the source. So when this button is set to unlinked, you can control the timeline playback using the buttons in the top left corner of the Timeline View window as you would with an audio player. If you are using Internal Timecode as a timecode source then these buttons will also operate the Internal Timecode source while linked.

![Timeline transport controls](/docs/images/Timeline-Timecode-Transport.png)

- \{ |< \} sets the live timecode back to zero.
- \{ |> \} starts playing from the start timecode.
- \{ > \} and \{ || \} are play / pause controls.
- \{ Stop \} stops playing and sets the live timecode back to zero.
- \{ Record \} enters live record mode but doesn't start timecode playback.

> When you are linked to a timecode (except Internal Timecode), the first four buttons are replaced by the timecode source name and only the Record button is shown.

Reviewer: I think the transport controls are also there for winamp. This will actually play/pause winamp too.

![Timeline external timecode controls](/docs/images/Timeline-Timecode-ExternalTransport.png)

If the timecode source is disabled in the **Timecode** menu, the time is shown in red.

![Timeline disabled](/docs/images/Timeline-Timecode-Disabled.png)

- If your timecode source starts at a big number, you can set a time offset to allow you to use zero-referenced times on your timeline.
See [Time Offset](./timelines/timeline-options.md#time-options).

The overview bar
------------------

The overview bar at the bottom of the screen gives you a view of the whole timeline from start to end. Triggers and markers are shown in miniature on the bar. You can make the main view zoom in and out by dragging the end markers of the bar to enclose a particular section. It's a good way of moving quickly around the timeline.

![Timeline overview bar](/docs/images/Timeline-Overview-Bar-Cropped.png)

Moving around the timeline view with the wheels
-----------------

> Using the context menu option \{Timeline Wheels\} you can set the wheels to control your view.

Reviewer: Might be worth a mention that this is only the case if you don't have any triggers selected

The wheel functions are
- Wheel A: Horizontal scroll
- Wheel B: Vertical scroll
- Wheel C: Zoom

Cursor
-----------------

The grey user cursor on the Timeline View sets the entry position for new triggers. It can also be used as a "play head" to 
manually move the play position along the timeline while testing. You can position the cursor by tapping the screen in the ruler
area at the top of the Timeline View.

The \{Select Cursor\} context menu option sets Wheel A to control the cursor position.

Selection tool types
------------------

The \{Arrow\} and \{Hand\} buttons let you choose different selection tools for editing.
- \{Arrow\} is for marquee editing, where you select items by drawing a marquee selection box around them (or direct clicking).
- \{Hand\} allows you to drag the timeline left and right or select by direct clicking.

You can also select the tool using the \{Tool Pan\}/\{Tool Select\} context menu button.

Table view
------------------

> You can switch the Timeline View to a table version which shows details for each trigger in time order, in the same format as a cue list. Buttons down the left hand side allow you to filter which tracks are shown in the list.

Reviewer: It's less switch the timeline view and more a totally separate view. You can have both on different screens at the same time for example. The selection is shared between the view

---
id: timelines
title: Timelines introduction
sidebar_label: Timelines introduction
---

> Timelines allow you to create a precisely timed light show which you can either

Author: Maybe just "sequence" would be OK. Trying to keep it short in the introduction.

run from the console's internal timer or synchronise to an external timecode
source. This can be useful for an opening sequence for a corporate
event, a complex song in a show where the song is pre-recorded or run to 
a click track or timecode, or a multimedia event run to a playback track. You
can also use it to create [complicated one-shot effects](./timelines/timeline-options.md#fader-tab) for busking.

> You can also create timed shows by creating timecoded cues in 
[cue-lists](./cue-lists/cue-list-timing.md#running-a-cue-list-to-timecode), but Timeline 
recording allows much easier creation and editing of a timed show.

Timelines can be built either from cues or chases you have already recorded, or by creating
new cues as you go along. You can record them in real time by operating the console
or by adding cues manually at specific times. You can think of a timeline as an automatic replay
system to push the faders and buttons on the console. The actions recorded in a timeline
are called **triggers**.

> To show the Timeline View window for a playback containing a timeline, touch the playback 

Reviewer 2: I think "Playback containing a timeline" might introduce a misleading concept.
```suggestion
To show the Timeline View window for a timeline, touch the playback 
```
"Playback information area" also seems a bit odd but so long as that is language used else where in the manual it's fine.

information area relating to the fader or press \<Open/View\> then the **select button** 
of the playback.

![Playback View of Timeline](/docs/images/Timeline-Window.png)

Triggers
------------------

To program a timeline you record actions at specific times
which can then be replayed by the console with the same timing. These
are called **Triggers**.

The following types of trigger can be recorded. You can record these actions in real time using **Live Record**
or enter them as manual triggers at specified times using the touch screen.

Trigger             | Action
--------------------|----------

> Set Level           | Set playback to a level over a fade time. This is like pushing up the fader over 2 seconds for example.

Reviewer 2: ```suggestion
Set Level           | Sets the fader level of a playback. This can be done over time. This applies whether or not the playback is physically assigned to a fader.
```

Something like that. I think it is important to emphasise that the playback fade time is a different thing.

Reviewer: or maybe "Sets the level of a playback over a time. This is similar to pushing up the fader over 2 seconds."

I think the example is helpful for understanding. It's also not technically setting the fader level, it's setting the playback level which is a different thing.

Author: What about "Fades up a playback to a set level over a time. This has the same effect as you pushing up a fader. If the playback has fade times programmed these also run."

Go to cue           | Go to a cue in a cue list. You can either specify a cue number or just record a *Go* trigger to run the next cue, which will follow the fade times programmed in the cue list.
Flash               | Flash a playback
Timed Flash         | Flash a playback with programmed fade times
Timed Flash and Go  | Timed flash with Go, for a cue list
Swop                | Swop a playback (cancels all other playbacks while active)
Preload             | Preload the LTP values of a playback over a number of seconds
Marker              | Mark a point in the timecode for information

> Wait for go         | Pause the timeline until you press Go to resume (only on internal timecode)

Reviewer 3: I believe you must be connected to the timeline for `<Go>` to work in this way (I could be wrong, can't check right now) - maybe we should mention this? I believe you can use `{Play}` if you're not connected.

example a win/walk-up sequence for an awards ceremony when you don't know how long the winner
will take to reach the stage. Shapes will continue to run while the timeline is paused.

Tracks
------------------

The timeline is split into **Tracks** which can be used to organise triggers into different groups or sections. Each different trigger in the timeline gets its own row in the track. 
- You can mute tracks to prevent them playing back using the \{Power\} button.
- You can lock tracks to prevent accidental changes to sections you've finished using the \{Lock\} button.
- Tracks can be collapsed so they take up less room on the screen, using the arrow button to the left of the track name.

Selecting timecode source
------------------

You can select one of the four available timecode sources to control your timeline from the [Timeline Options](./timelines/timeline-options.md#timecode-tab).

To set up and enable each of the timecode sources as Internal, MIDI, Clock or SMPTE use the \[Timecode\] softkey at the top level main menu.

Timecode linking and local timecode controls
------------------

> The \{Link\} button in the bottom left of the Timeline View window links or unlinks the timeline with the timecode source. Especially with external timecode, it's useful when recording and editing to be able to control the timecode yourself rather than having to continually ask the video guys to rewind the source. So when this button is set to unlinked, you can control the timeline playback using the buttons in the top left corner of the Timeline View window as you would with an audio player. 

Reviewer 2: There is more to life than video, for a start it would not be that ridiculous for the person running timecode to be a member of the band. Instead of guys, folks or videots can we say:

When using external timecode, it's useful to be able to temporarily control the timecode yourself when recording and editing if the external timecode source is not under your control.

- If you are using Internal Timecode or Winamp as a timecode source then these buttons will also operate the local Timecode source while linked (and will play/pause Winamp if that is the timecode source).

![Timeline transport controls](/docs/images/Timeline-Timecode-Transport.png)

> - \{ |< \} sets the live timecode back to zero.

Reviewer 3: ```suggestion
- \{ |< \} sets the live timecode back to the start time.
```
(I **believe** - maybe check this)

> - \{ |> \} starts playing from the start timecode.

Reviewer 2: As far as I am aware it starts playing from the cursor.

> - \{ |> \} starts playing from the start timecode.

Author: Seems to be from the cursor unless the cursor is before the start time, in which case it starts from the start time. The cursor seems to load at 0 when the console is started until changed by the user.

- \{ > \} and \{ || \} are play / pause controls.
- \{ Stop \} stops playing and sets the live timecode back to zero.
- \{ Record \} enters live record mode but doesn't start timecode playback.

When you are linked to a timecode (except Internal Timecode and Winamp), the first four buttons are replaced by the timecode source name and only the Record button is shown.

![Timeline external timecode controls](/docs/images/Timeline-Timecode-ExternalTransport.png)

> If the timecode source is disabled in the **Timecode** menu, the time is shown in red.

Reviewer 2: ```suggestion
If the timecode is disabled in the **Timecode** menu, the time is shown in red.
```
Timecode sources are not individually disabled or enabled. There is one options that rules them all.
